sim/rogue/rotation.go

```diff
@@ -245,17 +255,22 @@ func (rogue *Rogue) planRotation(sim *core.Simulation) []rogueRotationItem {
 }
 
 func (rogue *Rogue) setPriorityItems(sim *core.Simulation) {
+	mhDagger := rogue.Equip[proto.ItemSlot_ItemSlotMainHand].WeaponType == proto.WeaponType_WeaponTypeDagger
+	ohDagger := rogue.Equip[proto.ItemSlot_ItemSlotOffHand].WeaponType == proto.WeaponType_WeaponTypeDagger
 	rogue.Builder = rogue.SinisterStrike
 	rogue.BuilderPoints = 1
 	if rogue.Talents.Mutilate {
-		mhDagger := rogue.Equip[proto.ItemSlot_ItemSlotMainHand].WeaponType == proto.WeaponType_WeaponTypeDagger
-		ohDagger := rogue.Equip[proto.ItemSlot_ItemSlotOffHand].WeaponType == proto.WeaponType_WeaponTypeDagger
+
 		if mhDagger && ohDagger {
 			rogue.Builder = rogue.Mutilate
 			rogue.BuilderPoints = 2
 		}
 		rogue.setupAssassinationRotation(sim)
 	}
+	if rogue.Talents.Shadowstep {
```

Comment:
Instead of all these checks for specific talents, you can check which tree has the most points.

See warrior code for examples:
https://github.com/wowsims/wotlk/blob/master/sim/warrior/overpower.go#L81
https://github.com/wowsims/wotlk/blob/master/sim/warrior/warrior.go#L24